hail/python/hail/ir/matrix_ir.py

```diff
@@ -216,7 +217,7 @@ def __init__(self, child, new_row):
 
     def _compute_type(self):
         child_typ = self.child.typ
-        self.new_row._compute_type(child_typ.row_env(), child_typ.entry_env())
+        self.new_row._compute_type({**child_typ.row_env(), 'n_cols': hl.tint32}, child_typ.entry_env())
```

Comment:
This makes me nervous. How do we handle collisions here? What happens when someone uses `n_cols` as an identifier somewhere?

Reply:
there are no collisions -- this is treated like `'row'` is.

also, I think I need to rebase -- the PR that introduced 'n_rows' and 'n_cols' is already merged.